calculator.rb

```diff
@@ -0,0 +1,110 @@
+def print_operator_list(list)
+    count = 1
+    list.each do |key, value|
+        puts "#{count}. #{key}(#{value})"
+        count += 1
+    end
+end
+
+def get_result_type_from_user()
+    result_type = ["integer", "float"]
+    puts "Do you want your result to be integer or float? "
+    answer = gets.chomp.downcase
+    while !result_type.include?(answer)
+        print "Hey, that's not a valid option. Valid options are Integer or Float! "
+        answer = gets.chomp.downcase
+    end
+    return answer
+end
+
+def get_operator_input_from_user(operator_list)
+    operator = gets.chomp.downcase
+    until operator_list.keys.include?(operator.to_sym) || operator_list.values.include?(operator)
+        puts "Bummer, that's not a valid math operator. Please enter in a valid operator"
+        print_operator_list(operator_list)
+        operator = gets.chomp.downcase
+    end
+    
+    if operator == "add"
+        operator = '+'
+    elsif operator == "subtract"
+        operator = '-'
+    elsif operator == "multiply"
+        operator = '*'
+    elsif operator == "divide"
+        operator = '/'
+    elsif operator == "power"
+        operator = '^'
+    elsif operator == "mod"
+        operator = '%'
+    end
+    
+    return operator
+end
+
+def get_a_number_from_user()  
+    while true
+        input = gets.chomp
+        number =  Float(input) rescue false
```

Comment:
Fantastic solution! We'll get into details about `rescue` later in case you want to know more details about it, but this is a clever solution for this project